storybookjs/storybook
# Core: Add multiple DefinePlugin support for process.env

## Problem

Current storybook behavior is overriding the whole `process.env`. I'm trying to use it with https://github.com/7rulnik/isomorphic-env-webpack-plugin and it doesn't work because `process.env` already replaced by the storybook. So for now it's possible to find `DefinePlugin` in `webpackFinal` and replace `definitions` by hand.

## What I did

Currently `DefinePlugin` has definiton like this: 

```js
{
  'process.env': {...}
}
```

After this PR `DefinePlugin` will have:

```js
{
  'process.env.FOO': 'FOO_VALUE',
  'process.env.BAR': 'BAR_VALUE',
}
```

## How to test

- Is this testable with Jest or Chromatic screenshots? — No
- Does this need a new example in the kitchen sink apps? — No
- Does this need an update to the documentation? — No

Currently, we have a [custom snapshot serializer](https://github.com/KosyanMedia/storybook/blob/specific-process-env/lib/core/src/server/core-presets.test.ts#L86:L86) that outputs only plugins names.

But it should possible to set any env variable like before and override some env that doesn't set by pushing `DefinePlugin` in `webpackFinal`.
